### DISCUSSION TITLE
postgres user is no longer superuser?

### DISCUSSION THREAD

**Discussion by d-mok** [supabase/supabase#2495] [Questions] [answered]
Supabase gives us the `postgres` user of the postgresql database, so that we can login to the database using any client we want. For as long as I know, `postgres` is a superuser, so it bypass all permission check. But today, when I run this sql:
`SELECT * FROM auth.users`
It gives the error:
`SQL Error [42501]: ERROR: permission denied for table users`
This is abnormal, because I could run this sql in the past. I check the user `postgres` and find out that it is no longer a superuser, and it doesn't have the permission to select from the auth schema. I try to set `postgres` back to a superuser, but I have to be a superuser first to set a superuser ... so no luck.
This happens to my most recent Suapbase project. My other earlier projects don't see this issue.

I would like to know:
1. Is there anyone else encountering this problem? Or is it just me?
2. Did the Supabase team changed something causing this issue? Is it intended?
3. How can I set `postgres` back to a superuser? Or at least be granted permission on the auth schema?

**Comment by eifo** [comment, marked as answer]
Had same issue, run on supabase sql dashboard the following

alter user postgres with superuser

**Reply by d-mok** [reply]
Thanks! That fixes it for me.
But does anyone know the cause of this issue?

**Reply by mhaidarhanif** [reply]
Tried to do it via Supabase SQL editor:

```sql
alter user postgres with superuser;
```

Result is:

```
Success. No rows returned
```

Then tried to authenticate again using `postgres` with the same password.

Thanks! It works now. This just happens recently, not happening in last week. Supabase team might have changed something.

**Reply by fahimalizain** [reply]
Well, this doesn't work anymore. It fails with `Failed to run sql query: must be superuser to alter superuser roles or change superuser attribute` when executed from SQL Web Editor.

**Reply by sahrohit** [reply]
> Well, this doesn't work anymore. It fails with `Failed to run sql query: must be superuser to alter superuser roles or change superuser attribute` when executed from SQL Web Editor.

Anyone found a fix for this ?

**Reply by nick-barth** [reply]
Currently broken for me as well, would really appreciate it if we could get a response.

**Reply by Aijazmakerb** [reply]
did anyone found the solution,
need help asap

**Reply by GaryAustin1** [reply]
You really should go to the security thread to get views on your comments.
https://github.com/orgs/supabase/discussions/9314
This is a really old thread and about a slightly different case.

**Comment by sweatybridge** [comment]
For future reference, the hosted platform will default `postgres` user to non-superuser going forward. Users of existing projects will also be demoted to non-superuser. If this impacts your work, please refer to our [security patch](https://github.com/supabase/supabase/discussions/9314) for some common workarounds.

**Comment by opsb** [comment]
This is a blocker for using https://electric-sql.com/ with hosted Supabase.

**Comment by danielrosehill** [comment]
Just installed Supabase and encountering the same thing. This is a very unusual default posture. If I'm not mistaken, the password is specified only for user `postgres` so ... if that's not a superuser and users don't know the actual superusers' credentials ... what's the solution for taking admin-level actions on the DB?

**Reply by ardabeyazoglu** [reply]
You can login with supabase_admin for admin operations, but this is not enough for my case becaue i put some admin queries to migrations and supabase run them with "postgres" user.

**Reply by Ulises-Menem** [reply]
I need to change the password_encryption method, and i am not being able to do so

**Comment by ssteiger** [comment]
If anyone else runs into this issue: what worked for me was changing:
```bash
# SUPABASE_DB_USER="postgres"
SUPABASE_DB_USER="supabase_admin"
```
